packages/code-studio/src/dashboard/linker/Linker.tsx

```diff
@@ -29,15 +29,85 @@ import {
 } from '../events';
 import LayoutUtils from '../../layout/LayoutUtils';
 import LinkerOverlayContent from './LinkerOverlayContent';
-import LinkerUtils from './LinkerUtils';
-import LinkType from './LinkType';
+import LinkerUtils, { Link, LinkColumn, LinkType } from './LinkerUtils';
 import { PanelManager } from '../panels';
-import { UIPropTypes } from '../../include/prop-types';
 
 const log = Log.module('Linker');
 
-class Linker extends Component {
-  constructor(props) {
+export type PanelProps = {
+  glContainer: GoldenLayout.Container;
+  glEventHub: GoldenLayout.EventEmitter;
+};
+
+export type Panel = Component<PanelProps>;
+
+export type LinkFilterMapValue = {
+  columnType: string;
+  text: string;
+  value: unknown;
+};
+
+export type LinkFilterMap = Map<string, LinkFilterMapValue>;
+
+export type LinkDataMapValue = {
+  type: string;
+  text: string;
+  value: string;
+};
+
+export type LinkDataMap = Record<string, LinkDataMapValue>;
+
+export type LinkablePanel = Panel & {
+  setFilterMap: (filterMap: LinkFilterMap) => void;
+  unsetFilterValue: () => void;
+};
+
+export function isLinkablePanel(panel: Panel): panel is LinkablePanel {
+  return (panel as LinkablePanel).setFilterMap != null;
+}
+
+interface StateProps {
+  activeTool: string;
+  isolatedLinkerPanelId?: string;
+  links: Link[];
+  timeZone: string;
+}
+
+interface OwnProps {
+  layout: GoldenLayout;
+  panelManager: PanelManager;
+  localDashboardId: string;
+}
+
+const mapState = (state: LinkerState, ownProps: OwnProps) => ({
+  activeTool: getActiveTool(state),
+  isolatedLinkerPanelId: getIsolatedLinkerPanelIdForDashboard(
+    state,
+    ownProps.localDashboardId
+  ),
+  links: getLinksForDashboard(state, ownProps.localDashboardId),
+  timeZone: getTimeZone(state),
+});
+
+const connector = connect(mapState, {
```

Comment:
Also probably needs explicit return type (or actions moved to TS). This one may be fixed by `mapState` having an explicit return type. Or you can specify `connect<State>(...)` possibly